This is the 21 class version from Mikael based on TF2.0

tejaswid opened the pull request:
- This is the 21 class network from Mikael. Older network was 14 class. TODO: have a flag to choose this in the launch file.
- It is based on tensorflow 2.0 and Keras. Older network was based on tensorflow 1.0
- The octomap cropping is the exact same implementation in both branches.

MikaYeghi committed:
integrated network; depth_max_dist in launchfile; static cropping

MikaYeghi committed:
finished cropping

MikaYeghi committed:
integrated 21-class network (supply path to weights file .hdf5); bbx and octomap max dist in launchfile

MikaYeghi committed:
Remove last ~200 pixels

MikaYeghi committed:
integrated network properly

MikaYeghi committed:
integrated network properly

MikaYeghi committed:
finalized network; added launchfile description

tejaswid committed:
Added dependency on segmentation-models

The network is built upon models from the segmentation-models python module.

tejaswid committed:
Cleaned the code

tejaswid committed:
removed autosave file

tejaswid committed:
Num channels for network init is read from launch file

tejaswid committed:
disable occupancy-grid to increase fps and add launch file for running the code online on spot

(cherry picked from commit f638291072a63ae1dab3510e81fc8a513dfddbf3)

tejaswid committed:
Merge commit '345afe5b2da77b7d39a2911b2a70c6b693beede3' into td-cleanup

tejaswid committed:
Crop only when bounding box size is positive

tejaswid committed:
Resize color and depth images to label image size

benoit-robotics commented:
Can you explain the difference between this PR and this one : https://github.com/ori-drs/SegmentationMapping/pull/4 ?
Also I was reluctant to merge this one because it's unclear if the 21 class network is really better than the 14 class network. It's yet to be proved I think. And you cannot easily switch between the two networks with a parameter in a launch file.

tejaswid commented:
Sorry, I did not take a look at the PR you mentioned earlier. I checked it just now. Here are the differences.
- I cleaned the code a bit (removed absolute paths to user specific folders - solves the exact same issues you mention there)
- added missing dependencies to the readme file
- cherry picked the commits you made to master to run the code live on Spot
- added the checks on bounding box size that you implemented.

I agree that we need to test between 14 and 21 classes. I understand that is it nor straightforward to switch between the two settings from the launch file.

For the network part alone, I was planning on using the launch parameters to initialize the required network in the python script.
This will not work for the C++ script.

benoit-robotics commented:
> It is based on tensorflow 2.0 and Keras. Older network was based on tensorflow 1.0

What do you mean by that ?

benoit-robotics commented:
also there is a conflict to fix

tejaswid commented:
> > It is based on tensorflow 2.0 and Keras. Older network was based on tensorflow 1.0
> 
> What do you mean by that ?

The code in the master branch is running on compatibility mode of older tensorflow. [Here](https://github.com/ori-drs/SegmentationMapping/blob/master/scripts/segmentation_node#L33). 

This branch is completely in tensorflow 2.0.

benoit-robotics commented:
So it's ok if we don't merge this PR yet ? As we said I want to wait for the 21 class network to be better tested.

tejaswid commented:
Yes, let us leave it for now. I will run some tests and compare the performances this week.

tejaswid committed:
Fixed issue with timestamp of octomap.

It is set to timestamp of the depth image used to create the pointcloud

tejaswid committed:
dirty commit: switching to image transport pipeline

tejaswid committed:
added .py extension to python files

tejaswid committed:
Merge branch 'td-local' into td-cleanup

tejaswid committed:
added information about the map between 14 and 21 class networks

tejaswid committed:
reverted to without .py extension for consistency with master

tejaswid committed:
revert from grayscale to colour input

tejaswid committed:
added frontleft launch file

tejaswid committed:
Added label colors that match the network repo

tejaswid committed:
Color lebel image is published using cv. no PIL.

tejaswid committed:
added rosbag record script

tejaswid committed:
added indoor_state message and publisher

tejaswid committed:
Added new network model for internal realsense

tejaswid committed:
updated to use new model

tejaswid committed:
for spot - reverted back to skipping more frames

tejaswid committed:
rearranged config files
